algviz/picture/connection_map.py

```diff
@@ -0,0 +1,136 @@
+import svgutils.transform as svgutils
+import os
+
+from algviz.parser import structures
+from .anchor import Anchor
+from .svg_engine import SVGEngine
+
+"""Tracks connection points and connection pictures to be composed into a single picture"""
+
+pointers_temp_file = "temp_pointers.svg"
+
+class ConnectionMap:
+    class Connection:
+        def __init__(self, start_point, connect_pic, anchor):
+            self.start_point = start_point
+            self.connect_pic = connect_pic
+            if not connect_pic.is_drawn:
+                connect_pic.draw()
+            self.anchor = anchor
+
+            self.left_shift_y = 0
+            self.right_shift_y = 0
+
+        def is_pointer(self):
```

Comment:
I don't like this method name.
What does it mean if `is_pointer` is true?  Why do we want to know?  Can we ask a more direct question instead?  (If there's no better question, then we could be using `isinstance` instead of calling a method.  But I think you had something in mind.)